presto-main/src/main/java/com/facebook/presto/server/testing/TestingPrestoServer.java

```diff
@@ -385,6 +388,7 @@ public TestingPrestoServer(
             eventListenerManager = ((TestingEventListenerManager) injector.getInstance(EventListenerManager.class));
             clusterStateProvider = null;
             planCheckerProviderManager = injector.getInstance(PlanCheckerProviderManager.class);
+            expressionManager.loadExpressionOptimizerFactory();
```

Comment:
I feel like having an additional method to configure/load the expression optimizer is a bit of an anti-pattern. It creates the potential for bugs if someone is trying to properly create the expression manager by requiring this additional method to be called. I haven't dug much deeper, but is there a reason we can't perform the loading of the factory in the constructor of the `expressionManager`?

Reply:
I believe the reason for this convention is these loading methods might entail something expensive, or that could fail, and it's helpful to do such things outside of Guice's bootstrapping process.  Such failures are often very verbose because Guice returns the error in the context of where it failed during the bootstrapping process, and I think it's cleaner to do it outside of the Guice dependency graph so that the failure is clear and isolated.